devices/hgkg.js

```diff
@@ -11,15 +11,46 @@ module.exports = [
         model: 'BAC-002-ALZB',
         vendor: 'HKGK',
         description: 'BAC series thermostat',
-        fromZigbee: [fz.moes_thermostat],
-        toZigbee: [tz.moes_thermostat_child_lock, tz.moes_thermostat_current_heating_setpoint, tz.moes_thermostat_mode,
-            tz.hgkg_thermostat_standby, tz.moes_thermostat_sensor, tz.moes_thermostat_calibration,
-            tz.moes_thermostat_deadzone_temperature, tz.moes_thermostat_max_temperature_limit],
-        exposes: [e.child_lock(), e.deadzone_temperature(), e.max_temperature_limit(),
-            exposes.climate().withSetpoint('current_heating_setpoint', 5, 30, 1, ea.STATE_SET)
-                .withLocalTemperature(ea.STATE).withLocalTemperatureCalibration(-30, 30, 0.1, ea.STATE_SET)
-                .withSystemMode(['off', 'cool'], ea.STATE_SET).withRunningState(['idle', 'heat', 'cool'], ea.STATE)
-                .withPreset(['hold', 'program']).withSensor(['IN', 'AL', 'OU'], ea.STATE_SET)],
+        fromZigbee: [
+            fz.moes_thermostat,
+            fz.ignore_basic_report,
+            fz.ignore_tuya_set_time,
+        ],
+        onEvent: tuya.onEventSetLocalTime,
+        toZigbee: [
+            tz.moes_thermostat_child_lock,
+            tz.moes_thermostat_current_heating_setpoint,
+            tz.moes_thermostat_mode,
+            tz.hgkg_thermostat_standby,
+            tz.moes_thermostat_sensor,
+            tz.moes_thermostat_calibration,
+            tz.tuya_thermostat_schedule,
+            tz.tuya_thermostat_week,
+            tz.tuya_thermostat_schedule_programming_mode,
+            tz.tuya_thermostat_bac_fan_mode,
+        ],
+        exposes: [
+            e.switch(),
```

Comment:
Why was `e.switch()` added? I don't see any converter who can handle `state`.

Reply:
I'm able to switch 2 modes:
cooling (called preset_mode: hold)
heating (called preset_mode: program)
unable to switch to third fan_only mode (for some reason it has preset_mode: program state too when u switch it manually)

I think the issue comes up at tuya_thermostat - will fix it latter

fan_mode works ok - didn't work befor
current_heating_setpoint works ok as before
child_lock works ok as before
local_temperature works ok as before
system_mode works, but "cool" means on and "off" means off

Reply:
Can you remove the `e.switch()`? It makes the tests fails and thus is preventing the merge (since there is no converter for `state`)

Reply:
done, I have made linting